### ISSUE TITLE
[Bug] Cover art download impossible

### ISSUE THREAD

**Issue by bel3atar** [Fledge68/WiiFlow_Lite#229]
On a vWii connected to the internet, downloading cover art is not working
Using the latest version.
![ezgif com-gif-maker](https://user-images.githubusercontent.com/2218261/100650144-c53f0880-3343-11eb-8c24-aa8b88985612.gif)

**Comment by gingerbeardman** [comment]
Do you have a network set? Can other homebrew apps connect OK?

**Comment by bel3atar** [comment]
> homebrew

yes, I can download apps from the home brew browser

**Comment by gingerbeardman** [comment]
only the source code as it is a proprietary format that the covers are converted to.

did you try any recent older versions of WFL? I'm wondering if any of those work for you. there have been some changes in the networking code recently.

**Comment by Hakaisha** [comment]
You can place PNG format artwork in the 'boxcovers' and 'covers' folders. Wii/GC/WiiWare artwork should be named to match the Game ID, like 'GLZE99.png'. Plugin game covers should be named exactly after their game's file name, including the file extension. For example: 'Wizards & Warriors.nes.png'.

The boxcovers folder is for 3D artwork (front, back, spine). The covers folder is for 2D artwork (front only). 

The proprietary format that WiiFlow uses for covers is called WFC (WiiFlow Cache). These WFC files are created by WiiFlow using the PNG files that are placed in the aforementioned folders. The WFC files, once created, will be found in the Cache folder.

**Comment by bel3atar** [comment]
> did you try any recent older versions of WFL? I'm wondering if any of those work for you. there have been some changes in the networking code recently.

The bug appeared in version 5.4.0
Version 5.2.0 can correctly connect to GameTDB and fetch covers.

**Comment by Fledge68** [comment]
what version of wfl? 5.4.8?
do you use a proxy?

@wiidev can you look into this?

Ill try tomorrow on my wii u.

**Comment by bel3atar** [comment]
> what version of wfl? 5.4.8?
> do you use a proxy?
> 
> @wiidev can you look into this?
> 
> Ill try tomorrow on my wii u.

The bug exists in every version released after version 5.0.2. I tried all of them.

5.0.2 works fine

**Comment by gingerbeardman** [comment]
> You can place PNG format artwork in the 'boxcovers' and 'covers' folders. 

I did not know this!

**Comment by Fledge68** [comment]
I tried wfl 5.4.8 and my wii u this morning. connecting and downloading covers works fine - no problems.

**Comment by bel3atar** [comment]
how can we debug this further?

**Comment by Fledge68** [comment]
try renaming your wiiflow_lite.ini so wfl will create a new one. its located in dev:/apps/wiiflow.

if that doesn't work blackb0x recommends resetting your wii u wifi connection settings.

**Comment by bel3atar** [comment]
I switched to wifi and now it's working.
I was using the Wii ethernet <-> usb adapter before.

**Comment by bel3atar** [comment]
Switched back to ethernet and it's broken again.
Since the homebrew channel app works correctly with the ethernet adapter, this is obviously a bug in WiiFlow's side.

I wonder what method is used in HBC to detect a valid network connection.

**Comment by Fledge68** [comment]
interesting find.  i use wifi for my connection. wonder if there some other settings to check for the ethernet usb adapte. i really don't know. but you did say 5.2.0 does work with your ethernet usb adapter?

**Comment by bel3atar** [comment]
yes 5.2.0 works fine with the ethernet adapter, and it can correctly download from GameTDB.

**Comment by Fledge68** [comment]
I'll look between 5.20 and 5.40 for a commit that might have caused the difference but it will take some time as real life is in control at the moment.

**Comment by Fledge68** [comment]
could you go into main settings and startup settings and tell me if force cios is ON or OFF?
if it's on could you set it to OFF and try downloading with your ethernet adapter?

**Comment by bel3atar** [comment]
I have default settings
Force Load cIOS = Off
ForcecIOS Revision = Auto

**Comment by StrawberryBrains** [comment]
I also have this issue I'm afraid, I cannot connect to the internet via the Nintendo ethernet USB adapter when using Wiiflow. It works fine with other homebrew/software

**Comment by Fledge68** [comment]
So i'm wondering if this has anything to do with SD only option. Once set wiiflow never checks the USB Port.  The only USB controllers wiiflow has code for are ps3/ps4 controllers via libsickaxis.

**Comment by macromediaflash8** [comment]
I really hope this will be resolved soon... my ethernet port not being recognized is a pain.

**Comment by gingerbeardman** [comment]
> I really hope this will be resolved soon... my ethernet port not being recognized in this app is a pain.

Please post your debug output, as detailed earlier in this thread.

**Comment by macromediaflash8** [comment]
> Please post your debug output, as detailed earlier in this thread.

I'm sorry, but I don't understand, is it

>Please edit sd:/apps/wiiflow/wiiflow_lite.ini and set sd_write_log to yes. Repeat your tests. Attach sd:/wiiflow.log here.

or

>try renaming your wiiflow_lite.ini so wfl will create a new one. its located in dev:/apps/wiiflow.

**Comment by gingerbeardman** [comment]
@macromediaflash8 take a look at this comment: https://github.com/Fledge68/WiiFlow_Lite/issues/229#issuecomment-937459770

**Comment by macromediaflash8** [comment]
Here is my log file:
[wiiflow.log](https://github.com/Fledge68/WiiFlow_Lite/files/11011928/wiiflow.log)

**Comment by AdamCatalyst** [comment]
I’ve got this issue too, only started when the had to switch to Ethernet. Please let me know if I can provide anything to help get to the bottom of it.

**Comment by IShouldHaveBeenAQATester** [comment]
I've got this same issue with a wired adapter.  E-shop works in vWii. I haven't tried wifi as I don't have an SSID/AP that is setup with weak enough security to work.

**Comment by Trikzon** [comment]
I am still having this issue today. I can't use a wireless connection because I don't have access to a wireless network that works with the Wii. Ethernet works for everything else that I have tried that needs an internet connection

**Comment by Blayr** [comment]
> could you go into main settings and startup settings and tell me if force cios is ON or OFF? if it's on could you set it to OFF and try downloading with your ethernet adapter?

Turning off "Force Cios" is what fixed this for me.

**Comment by bosam** [comment]
> Turning off "Force Cios" is what fixed this for me.

Thank you! Even though my setting was OFF by default, I turned it ON and then OFF and I was able to download properly.

Same config as previously stated: vWii with the official Nintendo ethernet adapter
